lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -1551,4 +1565,12 @@ private function generateIdentifierName($identifier)
 
         return strtoupper(dechex(crc32($identifier->getName())));
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isValidIdentifier($name)
+    {
+        return !!preg_match('/^(?:(?:\p{L}|[_@#])(?:\p{L}|\p{Nd}|[$@#_])*|".*"|\[.*\])$/ui', $name);
```

Comment:
Sure thing, I can make that change tomorrow. As for the regex, it's checking for either a quoted string ([] or "") or a valid identifier as specified in the TSQL spec. The \p{L} is for unicode letters, etc.